### PR TITLE
Cell resize via ruler markers

### LINKED ISSUE #6688: Resize grid cells via ruler markers

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

It should be possible to resize grid cells using the ruler markers.

**Fix:**

When dragging a ruler marker:
1. start a new interaction for the new resize strategy variant which detects the closest cell on a given axis
2. freeze the existing markers so they don't move until the interaction is complete
3. show a pink snap line on the interaction target, with a label on top indicating the pin value
4. show a live offset line following the drag position

**Notes:**
- The snap line positioning might be tweaked in the future based on what we decide to do with "duplicate" ruler notches.
- The resize is only enabled for stretching grid items, same as we do with the regular resize. We should (probably?) incrementally add the ability to resize cells even if their child is not stretching when using the rulers.

**Manual Tests:**
I hereby swear that:

- [x] I opened a hydrogen project and it loaded
- [x] I could navigate to various routes in Play mode

Fixes #6688

### CODE REVIEW

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx [hunk 1]
```diff
@@ -680,7 +682,7 @@ export function combineApplicableControls(
   let gridControlsTargets: Map<WhenToShowControl, Array<GridIdentifier>> = new Map()
   for (const control of gridControlsInstances) {
     if (isGridControlsProps(control.props)) {
-      const possibleTargets = gridControlsTargets.get(control.show)
+      let possibleTargets = gridControlsTargets.get(control.show)
```
Comment (seanparsons) [inline editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx:685]: I don't think this is necessary?

File: editor/src/components/canvas/canvas-strategies/strategies/grid-resize-element-ruler-strategy.ts [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+      if (allCellBounds == null) {
+        return emptyStrategyApplicationResult
+      }
+
+      const bounds = getGridChildCellCoordBoundsFromCanvas(selectedElementMetadata, allCellBounds)
+      if (bounds == null) {
+        return emptyStrategyApplicationResult
+      }
+
+      const columns = allCellBounds[0]
```
Comment (seanparsons) [inline editor/src/components/canvas/canvas-strategies/strategies/grid-resize-element-ruler-strategy.ts:110]: Doesn't look like we bounds check this array.

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx (hunk 1 above)
Comment (ruggi) [reply editor/src/components/canvas/canvas-strategies/canvas-strategies.tsx:685]: see the push at line 689!

File: editor/src/components/canvas/canvas-strategies/strategies/grid-resize-element-ruler-strategy.ts (hunk 2 above)
Comment (ruggi) [reply editor/src/components/canvas/canvas-strategies/strategies/grid-resize-element-ruler-strategy.ts:114]: https://github.com/concrete-utopia/utopia/pull/6689/commits/c344a8439fecdcc914eb97a8a1ebd9c84acc553c

**Pull request merged**